src/ast/query.rs

```diff
@@ -661,6 +662,9 @@ pub enum TableFactor {
         args: Option<Vec<FunctionArg>>,
         /// MSSQL-specific `WITH (...)` hints such as NOLOCK.
         with_hints: Vec<Expr>,
+        /// Optional version qualifier to facilitate table time-travel, as
+        /// supported by BigQuery and MSSQL.
+        version: Option<TableVersion>,
```

Comment:
given it is so specific maybe this could be called `system_time: Option<SystemTime>` or something? I don't feel strongly

Reply:
Ideally, I imagine this field would evolve into something capable of capturing other qualifiers beyond time, e.g. something like explicit version numbers as [supported by delta-rs](https://docs.rs/deltalake/latest/deltalake/builder/enum.DeltaVersion.html). To that end, I'd vote to keep it more generic than `system_time`.

Reply:
Perhaps I could rename it to `temporal_qualifier: Option<TemporalQualifier>,`? Feels more SQL-y, and at the same time sufficiently generic such that it could also store explicit version number references (besides timestamps and intervals).